liblangutil/ErrorReporter.cpp

```diff
@@ -198,6 +198,30 @@ void ErrorReporter::typeError(SourceLocation const& _location, SecondarySourceLo
 	);
 }
 
+void ErrorReporter::typeError(SourceLocation const& _location, std::initializer_list<std::string const> const& _descriptions)
+{
+	std::string errorStr;
```

Comment:
Could we use `std::transform` here?

```
std::vector<std::string> allErrors;
std::transform(s.begin(), s.end(), std::back_inserter(allErrros), [](string error) -> std::string { return error; });
```

Reply:
how would that help?

Reply:
Aeh, you do not have to fiddle with iterators in a raw loop?

Reply:
but you're simply concatenating the strings in your code.. We need logic based on whether it's the last element or not to know if we should append a whitespace or not between the sentences..

Reply:
That could go in the lambda. I'm not sure which one would be more readable though.